gooddata-server-oauth2-autoconfigure/src/main/kotlin/CustomReactiveOAuth2UserService.kt

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2024 GoodData Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.gooddata.oauth2.server
+
+import mu.KotlinLogging
+import org.springframework.http.HttpStatus
+import org.springframework.security.oauth2.client.userinfo.DefaultReactiveOAuth2UserService
+import org.springframework.security.oauth2.client.userinfo.OAuth2UserRequest
+import org.springframework.security.oauth2.core.user.OAuth2User
+import org.springframework.web.server.ResponseStatusException
+import reactor.core.publisher.Mono
+
+class CustomReactiveOAuth2UserService : DefaultReactiveOAuth2UserService() {
+
+    val logger = KotlinLogging.logger {}
+
+    @Override
+    override fun loadUser(userRequest: OAuth2UserRequest): Mono<OAuth2User> {
```

Comment:
Is it possible to add some (unit) test?

Reply:
Had to adjust the code a bit, but now the tests are in place

Reply:
Looks good!